src/term/mod.rs

```diff
@@ -106,6 +106,7 @@ pub enum Term {
     nxt: Box<Term>,
   },
   Sup {
+    tag: Option<Name>,
```

Comment:
Rethinking this, we should probably make sup tags obligatory.
* If each dup and sup is given an unique label, the user-provided sups will never be annihilated.
* If dup and sup labels can overlap and are generated independently, we get behaviour that depends on other functions and that the user has no control. Depending on arbitrary decisions of the compiler, a sup may interact with dups that have equal or different labels, leading to different results.

So, to me the only sensible option is to force sups to have explicit labels